### ISSUE TITLE
[SC2086] No warning with `for file; do`

### ISSUE THREAD

**Issue by andlrc** [koalaman/shellcheck#795]
#### For bugs
- Rule Id SC2086
- My shellcheck version version: 0.4.5
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
for file
do
  sed : $file
done


```

#### Here's what shellcheck currently says:

{NOTHING}

#### Here's what I wanted or expected to see:

Double quote to prevent globbing and word splitting. [SC2086]

-----

It seems like `shellcheck` tries to be clever and will look what you are iterating in the `for` loop. But doesn't seems to realize potential errors wih `for stuff`

Writing`for stuff` is a shorthand for writing `for stuff in "$@"`. Currently `for stuff in "$@"` works as expected (Throws warning about pathname expansion and word splitting) while `for stuff` doesn't (No warning).

**Comment by koalaman** [comment]
Well caught! Fixed in 6ccb7e9
